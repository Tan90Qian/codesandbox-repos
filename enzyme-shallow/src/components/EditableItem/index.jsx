import React, { createContext, useContext } from "react";
import { Col, Form } from "antd";

export const Action = {
  view: "view",
  edit: "edit",
  cancel: "cancel",
  save: "save"
};

export const ModeContext = createContext(Action.view);

const FormItem = Form.Item;

export default function EditableItem(props) {
  const {
    label,
    viewContent = "暂无数据",
    className,
    style,
    children,
    disabled,
    span,
    ...restProps
  } = props;
  const mode = useContext(ModeContext);
  const defaultSpan = span ? { span } : { xs: 24, md: 12, lg: 8, xl: 6 };
  return (
    <Col {...defaultSpan} className={className} style={style} {...restProps}>
      <FormItem label={label}>
        {mode === Action.edit && !disabled ? children : viewContent}
      </FormItem>
    </Col>
  );
}
