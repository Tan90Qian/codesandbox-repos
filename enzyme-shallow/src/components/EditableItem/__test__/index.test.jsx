import React from 'react';
import { shallow } from 'enzyme';
import EditableItem, { Action, ModeContext } from '../index';

function ModeContextProvider(props) {
  const { children, mode } = props;
  console.log('mode provider', mode);
  return <ModeContext.Provider value={mode}>{children}</ModeContext.Provider>;
}

it('should render string', () => {
  const viewContent = 'test content';
  const wrapper = shallow(
    <EditableItem label="test" viewContent={viewContent}>
      test children
    </EditableItem>,
    {
      wrappingComponent: ModeContextProvider,
      wrappingComponentProps: {
        mode: Action.view,
      },
    }
  );
  expect(wrapper.getElement()).toMatchSnapshot();
  const provider = wrapper.getWrappingComponent();
  provider.setProps({
    mode: Action.edit,
  });
  wrapper.update();
  expect(wrapper.getElement()).toMatchSnapshot();
});
