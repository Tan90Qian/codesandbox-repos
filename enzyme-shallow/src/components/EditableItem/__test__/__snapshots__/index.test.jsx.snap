// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render string 1`] = `
<Col
  lg={8}
  md={12}
  xl={6}
  xs={24}
>
  <FormItem
    hasFeedback={false}
    label="test"
  >
    test content
  </FormItem>
</Col>
`;

exports[`should render string 2`] = `
<Col
  lg={8}
  md={12}
  xl={6}
  xs={24}
>
  <FormItem
    hasFeedback={false}
    label="test"
  >
    test content
  </FormItem>
</Col>
`;
